apps/root/src/services/aggregatorService.ts

```diff
@@ -113,25 +113,13 @@ export default class AggregatorService {
     let shouldValidate = !buyAmount && isOnNetwork;
 
     const network = chainId || currentNetwork.chainId;
-    let hasEnoughForSwap = true;
+    let hasEnoughForSwap = !!sellAmount;
+    const balance = await this.walletService.getBalance(from.address);
 
-    if (takerAddress && sellAmount) {
-      // const preAllowanceTarget = await this.sdkService.getAllowanceTarget();
-      // const allowance = await this.walletService.getSpecificAllowance(from, preAllowanceTarget);
-
-      // if (parseUnits(allowance.allowance, from.decimals).lt(sellAmount)) {
-      //   shouldValidate = false;
-      // }
-
-      if (shouldValidate) {
-        // If user does not have the balance do not validate tx
-        const balance = await this.walletService.getBalance(from.address);
-
-        if (balance.lt(sellAmount)) {
-          shouldValidate = false;
-          hasEnoughForSwap = false;
-        }
-      }
+    if (takerAddress && shouldValidate && sellAmount && balance.lt(sellAmount)) {
```

Comment:
this should check for protocol token or better yet, we have to fetch the balance of the token he wants to use